src/cli/commands/list-tests/formatters/tree.ts

```diff
@@ -0,0 +1,129 @@
+import path from "node:path";
+
+import type { TestCollection, TestDisabled } from "../../../../test-collection";
+import type { Suite, Test } from "../../../../types";
+
+type TreeSuite = {
+    id: string;
+    title: string;
+    line: number;
+    column: number;
+    suites: TreeSuite[];
+    // eslint-disable-next-line no-use-before-define
+    tests: TreeTest[];
+};
+
+type TreeTest = Omit<TreeSuite, "suites" | "tests"> & {
+    browserIds: string[];
+};
+
+type MainTreeRunnable = (TreeSuite | TreeTest) & {
+    file: string;
+};
+
+export const format = (testCollection: TestCollection): MainTreeRunnable[] => {
```

Comment:
here, in order to create the right tree, I move from the tests (which can be disabled) up the parent suites.